pyccl/halos/concentration.py

```diff
@@ -382,6 +382,171 @@ def _concentration(self, cosmo, M, a):
+                    self.b0 = 0.56
+                    self.b1 = 13.24
+                    self.c_alpha = 0.079
+                else:  # now it's 200c
+                    self.kappa = 1.79
+                    self.a0 = 2.15
+                    self.a1 = 2.06
+                    self.b0 = 0.88
+                    self.b1 = 9.24
+                    self.c_alpha = 0.51
+            else:  # fit all halos
+                if self.mdef.Delta == 'vir':
+                    self.kappa = 0.76
+                    self.a0 = 2.34
+                    self.a1 = 1.82
+                    self.b0 = 1.83
+                    self.b1 = 3.52
+                    self.c_alpha = -0.18
+                else:  # now it's 200c
+                    self.kappa = 1.10
+                    self.a0 = 2.30
+                    self.a1 = 1.64
+                    self.b0 = 1.72
+                    self.b1 = 3.60
+                    self.c_alpha = 0.32
+        else:  # use profile fitting method
+            if self.relaxed:  # fit only relaxed halos
+                if self.mdef.Delta == 'vir':
+                    self.kappa = 1.22
+                    self.a0 = 2.52
+                    self.a1 = 1.87
+                    self.b0 = 2.13
+                    self.b1 = 4.19
+                    self.c_alpha = -0.017
+                else:  # now it's either 200c or 500c
+                    if int(self.mdef.Delta) == 200:
+                        self.kappa = 0.60
+                        self.a0 = 2.14
+                        self.a1 = 2.63
+                        self.b0 = 1.69
+                        self.b1 = 6.36
+                        self.c_alpha = 0.37
+                    else:  # now it's 500c
+                        self.kappa = 0.38
+                        self.a0 = 1.44
+                        self.a1 = 3.41
+                        self.b0 = 2.86
+                        self.b1 = 2.99
+                        self.c_alpha = 0.42
+            else:  # fit all halos
+                if self.mdef.Delta == 'vir':
+                    self.kappa = 1.64
+                    self.a0 = 2.67
+                    self.a1 = 1.23
+                    self.b0 = 3.92
+                    self.b1 = 1.30
+                    self.c_alpha = -0.19
+                else:  # now it's either 200c or 500c
+                    if int(self.mdef.Delta) == 200:
+                        self.kappa = 1.19
+                        self.a0 = 2.54
+                        self.a1 = 1.33
+                        self.b0 = 4.04
+                        self.b1 = 1.21
+                        self.c_alpha = 0.22
+                    else:  # now it's 500c
+                        self.kappa = 1.10
+                        self.a0 = 2.30
+                        self.a1 = 1.64
+                        self.b0 = 1.72
+                        self.b1 = 3.60
+                        self.c_alpha = 0.32
+
+    def _dlsigmaR(self, cosmo, R, a, *, eps=1e-6, num=2):
```

Comment:
TODO: `num` is leftover from the previously-used `np.gradient` method. Remove it and replace it.